src/parallel/parallel.hpp

```diff
@@ -131,9 +132,13 @@ class SegmentLock {
 /// HELPER CLASSES
 ////////////////////////////////////////////////////////////////////////////////
 
-struct ParallelBucket {
+struct KeyValue {
     KeyType key = 0;
     ValueType value = 0;
+};
+
+struct ParallelBucket {
+    std::atomic<KeyValue> key_value;
     HashCodeType hashcode = 0;
```

Comment:
If we store the hashcode, does it need to be changed atomically with the key?

Are there any cases where changing the key but not the hashcode would result in an error?